### DISCUSSION TITLE
Interact with Intermediate Table Columns

### DISCUSSION THREAD

**Discussion by mefenlon** [statamic-rad-pack/runway#408] [Help]
The Title model has a Many to Many (Polymorphic) relationship to the model Projects. I am looking for a way to interact with the pivot table columns. Is this possible to do currently?

```
//Title Model
   public function projects(): MorphToMany
    {
        return $this->morphToMany(Project::class, 'projectable')
            ->withTimestamps()->withPivot('date', 'frontend_extra_attributes');
    }
```

I was think of approaching the problem by defining an intermediate table model, but interacting directly would bea real benefit.
https://laravel.com/docs/10.x/eloquent-relationships#defining-custom-intermediate-table-models

**Comment by duncanmcclean** [comment]
Unfortunately, Runway doesn't currently support pivot tables or `morphToMany` relationships. 

It may be possible to make pivot tables work at some point (I'd accept a PR if you can figure it out). 

However, I don't think we'd be able to make morph relationships work due to the fact a relationship could return different models which Statamic's relationships fieldtypes aren't built to support (give this FR an upvote: #245). Again, if anyone's able to figure out how to make this work, I'd accept a PR but I don't *think* it's possible.
